src/app/api/utils/featureFlags.ts

```diff
@@ -0,0 +1,26 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import appConstants from "../../../appConstants"
+
+type FeatureFlags = {
+  [key: string]: FeatureFlag | boolean
+}
+
+type FeatureFlag = {
+  name?: string
+  description?: string
+  dependencies?: [string]
+}
+
+const featureFlags = JSON.parse(appConstants.FEATURE_FLAGS) as FeatureFlags
+
+export function checkFeatureFlag(flag: string) : (FeatureFlag | boolean) {
```

Comment:
```suggestion
type FeatureFlags = Record<"test" | "otherFlag" | "otherFlag", FeatureFlag | boolean>;

type FeatureFlag = {
  name?: keyof FeatureFlags;
  description?: string;
  dependencies?: Array<keyof FeatureFlags>;
}

const featureFlags = JSON.parse(appConstants.FEATURE_FLAGS) as FeatureFlags

export function checkFeatureFlag(flag: keyof FeatureFlags) : (FeatureFlag | boolean) {
```

If we make the available flags an explicit union type of literal strings, it'll be easier to remove/change them later on than if they're arbitrary strings.

Reply:
@Vinnl The feature flags will be set via SRE in env vars (source of truth is whatever is set in the helm chart). The function here is purely to read that stringified JSON. I'm afraid that setting these restrictions would make it a lot more work than necessary (since everytime we change something in env var, we will need to remember to add that here)

I do like the keyof suggestion though

Reply:
> since everytime we change something in env var, we will need to remember to add that here

Only if we add a new feature flag, right? And only if we actually reference it in our code, so then we're modifying our code anyway. Or am I misunderstanding what you mean?

(I'd generally advise against enums, as they're a language-level rather than a type-level feature, and are incompatible with the proposed JS enums.)

Reply:
It doesn't matter if the flag is actually used in our code, as long as it exists in env var, it will need to be ref'd here since the key is strict. Yes should only apply when adding new feature flag (in env var)

I also think it gives the illusion that once you change it here, it's somehow automatically enabled. The real "change" needs to still happen with SRE

Reply:
Type checking only happens at build time, so our code wouldn't suddenly break if a new key gets added to the environment variable. You just wouldn't be able to write code that references that key without also updating the type.

The illusion is a good point, though the worst that will result in is that the feature flag wouldn't be enabled yet.

Reply:
Right.. hm I think another issue is there needs to be a tool that generates the json string for `.env` updates. I can already see this being a pretty annoying manual process. Anyways, I will commit this change